### DISCUSSION TITLE
Licenses / virtual items / BOM

### DISCUSSION THREAD

**Discussion by winzkigermany** [inventree/InvenTree#4490] [Q&A] [answered]
Hi, let's say I have a BOM with a part called "license". It's a license we have to buy *after* producing the BOM.
This part has a price, because I want it to be summed up in the BOM.
We don't buy it in advance and stock it somehere, so the stock level is normally 0

Now I want to issue a Build Order, but of course, the license part is out of stock.

What is the intended way to build such a product?
I don't understand the ramifications of virtual parts, consumables, or starting build orders when not all parts can be allocated.

I would have guessed, that making the "license part" a "virtual" part, and then let stock become negative, would be a great solution.
That way, I could restore stock of the virtual part to zero at the end of a month, by buying the "license part" from a supplier, in the number of "license parts" we used up during the month.
Can Inventree do such a thing?

Or does anybody else know how to handle this properly?
When I buy a "fake stock" of licenses, this results in a wrong (too high) inventory value.
When I leave out the licenses as a part in general, this results in a wrong BOM (too low) price calculation.
Letting stock of virtual parts go negative would be the simplest solution, I guess?

**Comment by matmair** [comment]
You can not allocate something that is not existing, I think implementing something like this would mostly lead to confusion.
I do not think that InvenTree is able to handle builds the way you want it to; not really sure if it should.

**Reply by winzkigermany** [reply]
After fiddling around some more hours, this is what I'm doing now:
In my BOM, I set the "consumable" flag on any "license part".
This gives me a complete BOM price calculation, including a price for the "license part".
And when creating a build order, consumables always allocate just fine, even if they're not in stock, so the production can begin. Yay!

What it does _not_ give me, is a count of the used "license part". If I want the number of license parts used, I need to count all the finished parts, that have my consumable "license part" in their BOM.

Although the solution is not perfect, I'm *very* happy that inventree lets me do this!

**Comment by SchrodingersGat** [comment]
@winzkigermany sounds like an interesting problem - if you do come up with a clean way of making it work with InvenTree please let us know

**Reply by winzkigermany** [reply]
Just did, see above.

Meanwhile, I also use a part "production cost" as a consumable in a BOM.
What I get from this, is the real cost of material and manufacturing, that goes into a part's BOM.
For this, I would have thought of "production cost" to be a "virtual part", but I think, I need to activate "internal pricing", to let it have a price in the BOM. I guess.

**Comment by winzkigermany** [comment, marked as answer]
To close this discussion, here are my solutions. Works for me.

In my BOM, I set the "consumable" flag on any "license part".
This gives me a complete BOM price calculation, including a price for the "license part".
And when creating a build order, consumables always allocate just fine, even if they're not in stock, so the production can begin. Yay!

I also use a part "production cost" as a consumable in a BOM.
What I get from this, is the real cost of material and manufacturing, that goes into a part's BOM.
For this, I would have thought of "production cost" to be a "virtual part", and I activate "internal pricing", to let it have a price in the BOM.
